otlp/trace.go

```diff
@@ -17,21 +19,34 @@ import (
 const (
 	traceIDShortLength = 8
 	traceIDLongLength  = 16
+	zeroSampleRate     = int32(0)
+	defaultSampleRate  = int32(1)
 )
 
-func TranslateHttpTraceRequest(body io.ReadCloser, ri RequestInfo) ([]map[string]interface{}, error) {
+type TranslateTraceRequestResult struct {
+	RequestSize int
+	Events      []Event
+}
+
+type Event struct {
+	Attributes map[string]interface{}
+	Timestamp  time.Time
+	SampleRate int32
+}
+
+func TranslateTraceRequestFromReader(body io.ReadCloser, ri RequestInfo) (*TranslateTraceRequestResult, error) {
 	if err := ri.ValidateHeaders(); err != nil {
 		return nil, err
 	}
 	request, err := parseOTLPBody(body, ri.ContentEncoding)
 	if err != nil {
 		return nil, ErrFailedParseBody
 	}
-	return TranslateGrpcTraceRequest(request)
+	return TranslateTraceRequest(request)
 }
 
-func TranslateGrpcTraceRequest(request *collectorTrace.ExportTraceServiceRequest) ([]map[string]interface{}, error) {
-	batch := []map[string]interface{}{}
+func TranslateTraceRequest(request *collectorTrace.ExportTraceServiceRequest) (*TranslateTraceRequestResult, error) {
```

Comment:
This function doesn't have an error return case

Reply:
Yeah, I went with providing an error in case we need to do some error handling in the future and then we won't need to change the API.